<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.13">
  <compounddef id="class_sil_kit_1_1_logic_error" kind="class" language="C++" prot="public">
    <compoundname>SilKit::LogicError</compoundname>
    <basecompoundref refid="class_sil_kit_1_1_sil_kit_error" prot="public" virt="non-virtual">SilKit::SilKitError</basecompoundref>
    <derivedcompoundref refid="class_sil_kit_1_1_length_error" prot="public" virt="non-virtual">SilKit::LengthError</derivedcompoundref>
    <derivedcompoundref refid="class_sil_kit_1_1_out_of_range_error" prot="public" virt="non-virtual">SilKit::OutOfRangeError</derivedcompoundref>
    <includes refid="exception_8hpp" local="no">exception.hpp</includes>
    <briefdescription>
<para>Error thrown as a consequence of faulty logic within the program such as violating logical preconditions or class invariants and may be preventable. </para>    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <inheritancegraph>
      <node id="158">
        <label>SilKit::LogicError</label>
        <link refid="class_sil_kit_1_1_logic_error"/>
        <childnode refid="159" relation="public-inheritance">
        </childnode>
      </node>
      <node id="159">
        <label>SilKit::SilKitError</label>
        <link refid="class_sil_kit_1_1_sil_kit_error"/>
        <childnode refid="160" relation="public-inheritance">
        </childnode>
      </node>
      <node id="160">
        <label>std::exception</label>
      </node>
      <node id="162">
        <label>SilKit::OutOfRangeError</label>
        <link refid="class_sil_kit_1_1_out_of_range_error"/>
        <childnode refid="158" relation="public-inheritance">
        </childnode>
      </node>
      <node id="161">
        <label>SilKit::LengthError</label>
        <link refid="class_sil_kit_1_1_length_error"/>
        <childnode refid="158" relation="public-inheritance">
        </childnode>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="163">
        <label>SilKit::LogicError</label>
        <link refid="class_sil_kit_1_1_logic_error"/>
        <childnode refid="164" relation="public-inheritance">
        </childnode>
      </node>
      <node id="164">
        <label>SilKit::SilKitError</label>
        <link refid="class_sil_kit_1_1_sil_kit_error"/>
        <childnode refid="165" relation="public-inheritance">
        </childnode>
      </node>
      <node id="165">
        <label>std::exception</label>
      </node>
    </collaborationgraph>
    <location file="/home/ubuntu/sil-kit/SilKit/include/silkit/participant/exception.hpp" line="96" column="1" bodyfile="/home/ubuntu/sil-kit/SilKit/include/silkit/participant/exception.hpp" bodystart="95" bodyend="98"/>
    <listofallmembers>
      <member refid="class_sil_kit_1_1_sil_kit_error_1a0077208e83054bce618c67ac98278a67" prot="protected" virt="non-virtual"><scope>SilKit::LogicError</scope><name>_what</name></member>
      <member refid="class_sil_kit_1_1_sil_kit_error_1adefb4bdec3000f1145f4c1f85281bdc7" prot="public" virt="non-virtual"><scope>SilKit::LogicError</scope><name>SilKitError</name></member>
      <member refid="class_sil_kit_1_1_sil_kit_error_1aa6cf5efd32d6d31bb2e7930ae7c82e83" prot="public" virt="non-virtual"><scope>SilKit::LogicError</scope><name>SilKitError</name></member>
      <member refid="class_sil_kit_1_1_sil_kit_error_1a15c8e944c8aafa377819ddbcdd34a2a3" prot="public" virt="non-virtual"><scope>SilKit::LogicError</scope><name>what</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
